src/date-picker/date-picker.jsx

```diff
@@ -30,31 +30,128 @@ const DatePicker = React.createClass({
+     * Follow this specificaction: ECMAScript Internationalization API 1.0 (ECMA-402).
+     */
     DateTimeFormat: React.PropTypes.func,
+
+    /**
+     * If true, automatically accept and close the picker on select a date.
+     */
     autoOk: React.PropTypes.bool,
+
+    /**
+     * Used to control how the DatePicker will be displayed when a user tries to set a date.
+     * `dialog` (default) displays the DatePicker as a dialog with a modal.
+     * `inline` displays the DatePicker below the input field (similar to auto complete)
+     */
     container: React.PropTypes.oneOf(['dialog', 'inline']),
+
+    /**
+     * This is the initial date value of the component.
+     * If either `value` or `valueLink` is provided they will override this
+     * prop with `value` taking precedence.
+     */
     defaultDate: React.PropTypes.object,
+
+    /**
+     * This function is called to format the date to display in the input box.
+     * By default, date objects are formatted to M/D/YYYY.
+     */
     formatDate: React.PropTypes.func,
+
+    /**
+     *
+     */
     hideToolbarYearChange: React.PropTypes.bool,
+
+    /**
+     * Locale used for formatting date. If you are not using the default value, you
+     * have to provide a DateTimeFormat that supports it. You can use Intl.DateTimeFormat
+     * if it's supported by your environment.
+     * https://github.com/andyearnshaw/Intl.js is a good polyfill.
+     */
     locale: React.PropTypes.string,
+
+    /**
+     * The ending of a range of valid dates. The range includes the endDate.
+     * The default value is current date + 100 years.
+     */
     maxDate: React.PropTypes.object,
+
+    /**
+     * The beginning of a range of valid dates. The range includes the startDate.
+     * The default value is current date - 100 years.
+     */
     minDate: React.PropTypes.object,
+
+    /**
+     * Tells the component to display the picker in portrait or landscape mode.
+     */
     mode: React.PropTypes.oneOf(['portrait', 'landscape']),
+
+    /**
+     * Callback function that is fired when the date value changes. Since there
+     * is no particular event associated with the change the first argument
+     * will always be null and the second argument will be the new Date instance.
+     */
     onChange: React.PropTypes.func,
+
+    /**
+     * Fired when the datepicker dialog is dismissed.
+     */
     onDismiss: React.PropTypes.func,
+
+    /**
+     * Callback function that is fired when the datepicker field gains focus.
+     */
     onFocus: React.PropTypes.func,
+
+    /**
+     * Fired when the datepicker dialog is shown.
+     */
     onShow: React.PropTypes.func,
+
+    /**
+     *
+     */
     onTouchTap: React.PropTypes.func,
+
+    /**
+     * Called during render time of a given day. If this method returns
+     * false the day is disabled otherwise it is displayed normally.
+     */
     shouldDisableDate: React.PropTypes.func,
+
+    /**
+     *
```

Comment:
@oliviertassinari If you follow the prop, it is passed down to date picker dialog. [Here](https://github.com/callemall/material-ui/blob/master/src/date-picker/date-picker-dialog.jsx#L183) it is passed down to calendar, but calendar has no such prop, but it has [disableYearSelection](https://github.com/callemall/material-ui/blob/master/src/date-picker/calendar.jsx#L41) which isn't called anywhere else. Thought it might be misnamed.